### PR TITLE
Convert ArgoCD Sync tests to ginkgo #123

### PR DESCRIPTION
**Pull request by Rizwana777**
#### Description:
- This PR is about converting ArgoCD Sync tests to ginkgo in cluster-agent
- Test files which are converted are :
           cluster-agent/utils/argocd_login_credentials_test.go
           cluster-agent/utils/argocd_terminate_app_command_test.go
           cluster-agent/utils/argocd_login_command_test.go
           cluster-agent/utils/argocd_sync_command_test.go

#### Link to JIRA Story (if applicable):
https://issues.redhat.com/browse/GITOPSRVCE-123)

### CODE REVIEW

File: cluster-agent/utils/argocd_login_command_test.go [hunk 1]
```diff
@@ -1,34 +1,31 @@
...
 	"github.com/redhat-appstudio/managed-gitops/cluster-agent/utils/mocks"
-	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/mock"
 )
 
-func TestArgoCDLoginCommand(t *testing.T) {
-
-	// Ensure we are able to login to get a token
-
-	t.Parallel()
```
Comment (jparsai) [inline cluster-agent/utils/argocd_login_command_test.go:16]: I guess you discussed it in meetings and IIRC there was some approach suggested to execute ginkgo tests in parallel, is that right?

Comment (Rizwana777) [reply cluster-agent/utils/argocd_login_command_test.go:16]: Yes @jparsai I discussed this and we need to pass "ginkgo -p"  command to run ginkgo tests parallel

Comment (jparsai) [reply cluster-agent/utils/argocd_login_command_test.go:16]: Then it would be better if you mention it somewhere, maybe a comment at the starting of functions to inform others that these tests are expected to be executed parallelly.

Comment (Rizwana777) [reply cluster-agent/utils/argocd_login_command_test.go:16]: Yes @jparsai  added comment

**Comment by Rizwana777** [comment]
@jparsai  Thank you for the review, updated changes , PTAL

File: cluster-agent/utils/argocd_login_command_test.go (hunk 1 above)
Comment (jparsai) [reply cluster-agent/utils/argocd_login_command_test.go:16]: Is CI (PR Build) is also running it in parallel?

File: cluster-agent/utils/argocd_login_credentials_test.go [hunk 2]
```diff
@@ -2,118 +2,108 @@ package utils
...
+				mockClient: mockAppClient,
+			}
+
+			cs := NewCredentialService(&clientGenerator, true)
+			Expect(err).To(BeNil())
+
+			creds, argoClient, err := cs.GetArgoCDLoginCredentials(context.Background(), "openshift-gitops", "12", true, k8sClient)
+			Expect(err).To(BeNil())
+			Expect(argoCDCredentials{} == creds).To(BeFalse())
+			Expect(argoClient).ToNot(BeNil())
```
Comment (jparsai) [inline cluster-agent/utils/argocd_login_credentials_test.go:90]: You are supposed to check for empty object here as well.

**Comment by jparsai** [comment]
@Rizwana777 it LGTM, just need one minor change.

File: cluster-agent/utils/argocd_login_credentials_test.go (hunk 2 above)
Comment (Rizwana777) [reply cluster-agent/utils/argocd_login_credentials_test.go:90]: For argoClient ? @jparsai

File: cluster-agent/utils/argocd_login_command_test.go (hunk 1 above)
Comment (Rizwana777) [reply cluster-agent/utils/argocd_login_command_test.go:16]: I don't think so @jparsai

File: cluster-agent/utils/argocd_login_credentials_test.go (hunk 2 above)
Comment (jparsai) [reply cluster-agent/utils/argocd_login_credentials_test.go:90]: Yes, in existing test I see empty check for argoClient as well.
https://github.com/redhat-appstudio/managed-gitops/blob/2fe8209b9711418db191393fc601eea2fbd6f2df/cluster-agent/utils/argocd_login_credentials_test.go#L102

Comment (Rizwana777) [reply cluster-agent/utils/argocd_login_credentials_test.go:90]: argoClient is "var argoClient argocdclient.Client"  here argocdclient.Client is an interface, so we cannot check for empty object @jparsai

Comment (jparsai) [reply cluster-agent/utils/argocd_login_credentials_test.go:90]: Okay, I see in case of failure `GetArgoCDLoginCredentials` returns `nil` for `argoClient`, so I guess no need to check for empty like existing test just check for `nil` is enough.
https://github.com/redhat-appstudio/managed-gitops/blob/2fe8209b9711418db191393fc601eea2fbd6f2df/cluster-agent/utils/argocd_login_credentials.go#L59

Comment (Rizwana777) [reply cluster-agent/utils/argocd_login_credentials_test.go:90]: Yup @jparsai  then there is no need to change this "Expect(argoClient).ToNot(BeNil())" right ?

**Review by jgwest — APPROVED** [review]
Looks great, thanks @Rizwana777 , and thanks for reviewing @jparsai !
